path_collector_test.go

```diff
@@ -449,3 +453,86 @@ func useDoublestarCollector(tc testCase, path string) yamlfmt.PathCollector {
 		Exclude: tc.excludePatterns.allPatterns(path),
 	}
 }
+
+func TestPatternFile(t *testing.T) {
+	t.Parallel()
+
+	makePatterns := func(patterns ...string) []byte {
+		var b bytes.Buffer
+
+		fmt.Fprintln(&b, "# Comment followed by empty line")
+		fmt.Fprintln(&b)
+		for _, p := range patterns {
+			fmt.Fprintln(&b, p)
+		}
+
+		return b.Bytes()
+	}
+
+	cases := []struct {
+		name      string
+		patterns  []byte
+		haveFiles []string
+		wantFiles []string
+	}{
+		{
+			name:      "yaml and yml files",
+			patterns:  makePatterns("*.yaml", "*.yml"),
+			haveFiles: []string{"x.yaml", "y.yml", "README.md"},
+			wantFiles: []string{"x.yaml", "y.yml"},
+		},
+		{
+			name:      "ignore pattern",
+			patterns:  makePatterns("*.yaml", "*.yml", "!test_input.yaml"),
+			haveFiles: []string{"x.yaml", "y.yml", "test_input.yaml"},
+			wantFiles: []string{"x.yaml", "y.yml"},
+		},
+		{
+			name:      "descent into directories",
+			patterns:  makePatterns("*.yaml", "*.yml"),
+			haveFiles: []string{"a/x.yaml", "b/y.yml"},
+			wantFiles: []string{"a/x.yaml", "b/y.yml"},
+		},
+		{
+			name:      "exclude directories",
+			patterns:  makePatterns("*.yaml", "*.yml", "!a/"),
+			haveFiles: []string{"a/x.yaml", "b/y.yml"},
+			wantFiles: []string{"b/y.yml"},
+		},
+		{
+			name:      "matches are rooted at the working directory",
+			patterns:  makePatterns("*.yaml", "!/x.yaml"),
+			haveFiles: []string{"x.yaml", "a/x.yaml"},
+			wantFiles: []string{"a/x.yaml"},
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
```

Comment:
Since this needs to be back-compat to Go 1.18 for now, you'll need to include `tc := tc` above this for it to work with the `t.Parallel()`.

Reply:
✅ done

Great catch, thanks!